smart-contract-verifier/smart-contract-verifier/src/lookup_methods/find_methods.rs

```diff
@@ -18,9 +18,68 @@ pub struct LookupMethodsResponse {
     pub methods: BTreeMap<String, Method>,
 }
 
+pub fn find_methods_from_compiler_output(
+    contract_name: &String,
+    output: &CompilerOutput,
+) -> anyhow::Result<LookupMethodsResponse> {
```

Comment:
I'm not sure if that would be easier (especially considering that parsing of CompilerOutput is already written), as an option, we may also get all required data from the `compilation_artifacts`, `deployed_code_artifacts` and `local_bytecode_parts`

https://github.com/blockscout/blockscout-rs/blob/main/smart-contract-verifier/smart-contract-verifier-server/tests/test_cases_solidity/no_metadata_hash.json